src/TLuaInterpreter.cpp

```diff
@@ -10201,86 +10274,206 @@ int TLuaInterpreter::getCmdLine(lua_State* L)
+            // loaded - this will produce an additinoal message even though
+            // we were successful - so report it (or any other future message,
+            // like, say, an MD5 hash of the contents - a future idea!)
+            lua_pushfstring(L, "Note, during installation of package \"%s\": %s.",
+                            packageName.toUtf8().constData(), errorMessage.toUtf8().constData());
+            return 2;
+        }
+    } else {
+        lua_pushnil(L);
+        lua_pushfstring(L, "Failed to install package \"%s\", reason: %s.",
+                        packageName.toUtf8().constData(), errorMessage.toUtf8().constData());
+        return 2;
+    }
 }
 
 int TLuaInterpreter::uninstallPackage(lua_State* L)
 {
-    string event;
+    QString packageName;
     if (!lua_isstring(L, 1)) {
-        lua_pushstring(L, "uninstallPackage(): wrong argument type");
-        lua_error(L);
-        return 1;
+        lua_pushfstring(L, "uninstallPackage: bad argument #1 type (package name as string expected, got %s!)",
+                        luaL_typename(L, 1));
+        return lua_error(L);
     } else {
-        event = lua_tostring(L, 1);
+        packageName = QString::fromUtf8(lua_tostring(L, 1));
+        if (packageName.isEmpty()) {
+            lua_pushnil(L);
+            lua_pushstring(L, "An empty string is not a valid name for a package to uninstall.");
+            return 2;
+        }
     }
+
     Host& host = getHostFromLua(L);
-    QString package = event.c_str();
-    host.uninstallPackage(package, 0);
-    return 0;
+    QString errorMessage;
+    if (host.uninstallPackage(packageName, 0, &errorMessage)) {
+        // The above will also update the package manager if it happens to be
+        // open so we (no longer) have to check for and do that here.
+        lua_pushboolean(L, true);
+        if (!errorMessage.isEmpty()) {
+            // Under some circumstances we may conceivable get a message even
+            // if succesful:
+            lua_pushfstring(L, "Note, during uninstallation of package \"%s\": %s.",
+                            packageName.toUtf8().constData(), errorMessage.toUtf8().constData());
+            return 2;
+        } else {
+            return 1;
+        }
+    } else {
+        lua_pushnil(L);
+        lua_pushfstring(L, "Failed to uninstall package \"%s\", reason: %s.",
+                        packageName.toUtf8().constData(), errorMessage.toUtf8().constData());
+        return 2;
+    }
 }
 
 int TLuaInterpreter::installModule(lua_State* L)
 {
-    string modName;
+    QString moduleName;
+    bool isToSync = false;
     if (!lua_isstring(L, 1)) {
-        lua_pushstring(L, "installModule: wrong first argument (should be a path to module)");
-        lua_error(L);
-        return 1;
+        lua_pushfstring(L, "installModule: bad argument #1 type (module {path and/or} file name as string expected, got %s!)",
```

Comment:
Same as above

Reply:
I've gone for:

     module file name {may include a path}